src/pages/index.js

```diff
@@ -102,33 +197,104 @@ function renderCard(data) {
 }
 
 function getCardElement(cardData) {
-    const card = new Card(cardData, '#card-template', handleImageClick);
+    console.log(cardData);
+    const card = new Card(cardData, '#card-template', handleImageClick, handleDeleteButtonSubmit, handleCardLike, handleCardDislike);
     return card.getView();
 }
 
-function handleImageClick(cardData) {
-    popupWithImage.open(cardData);
+function showErrorMessage(err){
+    const messageElement = document.querySelector('.modal__button');
+    messageElement.textContent = `Error: ${err}`;
 }
 
 /* -------------------------------------------------------------------------- */
 /*                               Event Handlers                               */
 /* -------------------------------------------------------------------------- */
 
+function handleImageClick(cardData) {
+    popupWithImage.open(cardData);
+}
+
+function handleCardLike(card){
+    api
+        .likeCard(card._id)
+        .then(() => {
+            card.cardIsLiked(true);
+        })
+        .catch((err) => {
+            console.log(err);
+        });
+}
+
+function handleCardDislike(card) {
+    api
+        .dislikeCard(card._id)
+        .then(() => {
+            card.cardIsLiked(false);
+        })
+        .catch((err) => {
+            console.error(err);
+        });
+}
+
 function handleProfileEditSumbit (userData) {
-    console.log(userData);
     const name = userData.name;
     const job = userData.bio;
-    userInfo.setUserInfo(name, job);
-    popupEditForm.close();
+    console.log(name);
+    console.log(job);
+    editPostLoading.editPost();
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```